plutus-benchmark/bls-benchmarks/src/PlutusBenchmark/BLS12_381/Common.hs

```diff
@@ -0,0 +1,349 @@
+type UProg   = UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
+type UDBProg = UPLC.Program UPLC.DeBruijn      DefaultUni DefaultFun ()
+
+
+compiledCodeToTerm
+    :: Tx.CompiledCodeIn DefaultUni DefaultFun a -> UTerm
+compiledCodeToTerm (Tx.getPlcNoAnn -> UPLC.Program _ _ body) = body
+
+{- | Remove the textual names from a NamedDeBruijn program -}
+toAnonDeBruijnProg :: UProg -> UDBProg
+toAnonDeBruijnProg (UPLC.Program () ver body) =
+    UPLC.Program () ver $ UPLC.termMapNames (\(UPLC.NamedDeBruijn _ ix) -> UPLC.DeBruijn ix) body
+
+-- Create a list containing n bytestrings of length l.  This could be better.
+listOfSizedByteStrings :: Integer -> Integer -> [ByteString]
+listOfSizedByteStrings n l = unsafePerformIO . G.sample $
+                             G.list (R.singleton $ fromIntegral n)
+                                  (G.bytes (R.singleton $ fromIntegral l))
+
+---------------- Examples ----------------
+
+-- Hash some bytestrings onto G1 and add them all together
+
+{-# INLINABLE hashAndAddG1 #-}
+hashAndAddG1 :: [BuiltinByteString] -> BuiltinBLS12_381_G1_Element
+hashAndAddG1 [] = error ()
+hashAndAddG1 (p:ps) =
+    go ps (Tx.bls12_381_G1_hashToGroup p)
+    where go [] acc     = acc
+          go (q:qs) acc = go qs $ Tx.bls12_381_G1_add (Tx.bls12_381_G1_hashToGroup q) acc
+
+mkHashAndAddG1Script :: [ByteString] -> UProg
+mkHashAndAddG1Script l =
+    let points = map toBuiltin l
+    in Tx.getPlcNoAnn $ $$(Tx.compile [|| hashAndAddG1 ||]) `Tx.unsafeApplyCode` Tx.liftCode points
+
+-- Hash some bytestrings onto G2 and add them all together
+{-# INLINABLE hashAndAddG2 #-}
+hashAndAddG2 :: [BuiltinByteString] -> BuiltinBLS12_381_G2_Element
+hashAndAddG2 [] = error ()
+hashAndAddG2 (p:ps) =
+    go ps (Tx.bls12_381_G2_hashToGroup p)
+    where go [] acc     = acc
+          go (q:qs) acc = go qs $ Tx.bls12_381_G2_add (Tx.bls12_381_G2_hashToGroup q) acc
+
+mkHashAndAddG2Script :: [ByteString] -> UProg
+mkHashAndAddG2Script l =
+    let points = map toBuiltin l
+    in Tx.getPlcNoAnn $ $$(Tx.compile [|| hashAndAddG2 ||]) `Tx.unsafeApplyCode` Tx.liftCode points
+
+-- Deserialise a list of compressed G1 points and add them all together
+{-# INLINABLE uncompressAndAddG1 #-}
+uncompressAndAddG1 :: [BuiltinByteString] -> BuiltinBLS12_381_G1_Element
+uncompressAndAddG1 [] = error ()
+uncompressAndAddG1 (p:ps) =
+    go ps (Tx.bls12_381_G1_uncompress p)
+    where go [] acc     = acc
+          go (q:qs) acc = go qs $ Tx.bls12_381_G1_add (Tx.bls12_381_G1_uncompress q) acc
+
+mkUncompressAndAddG1Script :: [ByteString] -> UProg
+mkUncompressAndAddG1Script l =
+    let points = map (Tx.bls12_381_G1_compress . Tx.bls12_381_G1_hashToGroup . toBuiltin) l
+    in Tx.getPlcNoAnn $ $$(Tx.compile [|| uncompressAndAddG1 ||]) `Tx.unsafeApplyCode` Tx.liftCode points
+
+
+-- Check that point addition is commutative in G1
+checkUncompressAndAddG1_Haskell :: Integer -> IO ()
```

Comment:
shouldn't this be in the plutus-core test suite if anywhere?

Reply:
I think I left that in there by accident.  I've removed it and the other one.